sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -152,6 +155,25 @@ export class AssessmentControllerService {
     return disbursementDetails;
   }
 
+  /**
+   * Calculate the sum of 'valueAmount' for each DisbursementValue within an array of DisbursementSchedules.
+   * @param disbursementSchedules disbursement schedule details.
+   * @returns The total sum of 'valueAmount' across all DisbursementSchedules.
+   */
+  private sumDisbursementValueAmounts(
+    disbursementSchedules: DisbursementSchedule[],
+  ): number {
+    return disbursementSchedules.reduce((totalValueAmount, schedule) => {
```

Comment:
The logic could be simplified like this.

```ts
 return disbursementSchedules
    .flatMap((disbursement) => disbursement.disbursementValues)
    .map((disbursementValue) => disbursementValue.valueAmount)
    .reduce((accumulator, currentValue) => accumulator + currentValue, 0);
```

Reply:
I agree, it can be done in many different ways, but i believe for nested arrays scenarios and summing a value, i believe reduce is generally more efficient and appropriate. Also i thought, of creating a private method to have the return in the getAssessmentNOA method cleaner. 
Do you still believe this needs to be changed?

Reply:
I don't have any hard feelings but I like Dheepak's suggestion cleaner.

Reply:
@guru-aot I still feel that it would make the code cleaner and easy to grasp.

Reply:
We have the same approach suggested by @dheepak-aot in other places and it was exactly the same that I mentioned over our call about the ticket @guru-aot maybe it was not clear during that time.
Please apply the suggestion since it seems cleaner.